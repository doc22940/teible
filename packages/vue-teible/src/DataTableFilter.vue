<template>
  <div :class="theme.datatable__filter">
    <input
      :value="filter" type="text" :class="theme.datatable__input"
      placeholder="Filter table data" @input="update($event.target.value)"
    >
    <div v-if="filter" :class="theme.datatable__clear" @click.stop="clear">
      <a :class="theme.datatable__x" href="#" @click.stop.prevent="clear">Ã—</a>
    </div>
  </div>
</template>
<script>
export default {
  name: 'DataTableFilter',
  props: {
    filter: {
      type: String,
      required: true
    }
  },
  computed: {
    theme () {
      return this.$theme()
    }
  },
  methods: {
    update (filter) {
      this.$emit('update:filter', filter)
    },
    clear () {
      this.$emit('update:filter', '')
    }
  },
  inject: ['$theme']
}
</script>
