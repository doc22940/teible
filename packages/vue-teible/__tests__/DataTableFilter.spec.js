import { shallowMount } from '@vue/test-utils'
import idObj from 'identity-obj-proxy'
import DataTableFilter from '../src/DataTableFilter.vue'

describe('DataTableFilter', () => {
  it(`renders correctly`, () => {
    let wrapper = shallowMount(DataTableFilter, {
      propsData: {
        filter: 'default'
      },
      provide: () => ({
        $theme: () => idObj
      })
    })
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.find('input').element.value = 'new value'
    wrapper.find('input').trigger('input')
    expect(wrapper.emitted()['update:filter']).toEqual([['new value']])

    wrapper.find('.datatable__x').trigger('click')
    expect(wrapper.emitted()['update:filter']).toEqual([['new value'], ['']])
  })
})
