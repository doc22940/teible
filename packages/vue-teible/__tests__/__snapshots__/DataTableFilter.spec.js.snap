// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableFilter renders correctly 1`] = `
<div class="datatable__filter"><input type="text" placeholder="Filter records" class="datatable__input">
  <div class="datatable__clear"><a href="#" class="datatable__x">Ã—</a></div>
</div>
`;
